cpp/src/interop/from_arrow_device.cu

```diff
@@ -88,6 +93,310 @@ data_type arrow_to_cudf_type(const ArrowSchemaView* arrow_view)
+  CUDF_CUDA_TRY(cudaMemcpyAsync(data.data(),
+                                reinterpret_cast<uint8_t const*>(data_buffer),
+                                buffer_length,
+                                cudaMemcpyDefault,
+                                stream.value()));
+  auto out_col = mask_to_bools(static_cast<bitmask_type*>(data.data()),
+                               input->offset,
+                               input->offset + input->length,
+                               stream,
+                               mr);
+
+  auto const has_nulls = skip_mask ? false : input->buffers[validity_buffer_idx] != nullptr;
+  if (has_nulls) {
+    auto out_mask = detail::copy_bitmask(static_cast<bitmask_type*>(get_mask_buffer(input)->data()),
+                                         input->offset,
+                                         input->offset + input->length,
+                                         stream,
+                                         mr);
+
+    out_col->set_null_mask(std::move(out_mask), input->null_count);
+  }
+
+  return out_col;
+}
+
+template <>
+std::unique_ptr<column> dispatch_copy_from_arrow_device::operator()<cudf::string_view>(
+  ArrowSchemaView* schema, ArrowArray const* input, data_type type, bool skip_mask)
+{
+  if (input->length == 0) { return make_empty_column(type_id::STRING); }
+
+  const void* offset_buffers[2] = {nullptr, input->buffers[fixed_width_data_buffer_idx]};
+  ArrowArray offsets_array      = {
+         .length     = input->offset + input->length + 1,
+         .null_count = 0,
+         .offset     = 0,
+         .n_buffers  = 2,
+         .n_children = 0,
+         .buffers    = offset_buffers,
+  };
+
+  size_type const char_data_length =
+    reinterpret_cast<int32_t const*>(offset_buffers[1])[input->length + input->offset];
+  const void* char_buffers[2] = {nullptr, input->buffers[2]};
+  ArrowArray char_array       = {
+          .length     = char_data_length,
+          .null_count = 0,
+          .offset     = 0,
+          .n_buffers  = 2,
+          .n_children = 0,
+          .buffers    = char_buffers,
+  };
+
+  nanoarrow::UniqueSchema offset_schema;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaInitFromType(offset_schema.get(), NANOARROW_TYPE_INT32));
+
+  nanoarrow::UniqueSchema char_data_schema;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaInitFromType(char_data_schema.get(), NANOARROW_TYPE_INT8));
+
+  ArrowSchemaView view;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, offset_schema.get(), nullptr));
+  auto offsets_column =
+    this->operator()<int32_t>(&view, &offsets_array, data_type(type_id::INT32), true);
+  NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, char_data_schema.get(), nullptr));
+  auto chars_column = this->operator()<int8_t>(&view, &char_array, data_type(type_id::INT8), true);
+
+  auto const num_rows = offsets_column->size() - 1;
+  auto bitmask        = get_mask_buffer(input);
+  auto out_col        = cudf::column_view(data_type{type_id::STRING},
```

Comment:
Sorry about this but I think your original implementation was better than my suggestion here.
I was thinking the data was already copied to the device and we could just wrap it with a `column_view` but the copy of the offsets, bitmask, etc need to occur and now those are double-copied because of my suggestion.
The original implementation only did the double-copy if the offset!=0 and so I'd like to return to that implementation.
Sorry again.

Reply:
updated back to the original implementation